app/models/reducers/count_choices_reducer.rb

```diff
@@ -0,0 +1,13 @@
+module Reducers
+  class CountChoicesReducer < Reducer
+    attr_reader :sub_ranges
+
+    def reduction_data_for(extractions)
+      ReductionResults.build do |results|
+        extractions.each do |extraction|
```

Comment:
@camallen `ReductionResults.build` is just `ReductionResults.new.tap { ... }.to_h` with a nicer API. Not a loop. But yeah, there are quite a few `end`s in at the end of the file, so I see what you mean.